.github/workflows/release.yml

```diff
@@ -0,0 +1,110 @@
+        # This sets the utility environment variable 'MAVEN_VERSION', that contains the version from the pom.xml
+        LOCAL_MAVEN_VERSION=$(mvn org.apache.maven.plugins:maven-help-plugin:3.1.0:evaluate -Dexpression=project.version -q -DforceStdout)
+        echo "Collected the pom.xml version. The version is '$LOCAL_MAVEN_VERSION'"
+        # Replace the '${project.version}' variable in the plugin.yml with the maven version and build number to mark this as offical snapshot build
+        sed -i 's~\${maven-version}~$LOCAL_MAVEN_VERSION~g' ./BetonQuest-core/src/main/resources/plugin.yml
+        echo "Replaced the variable '\${project.version}' with '$LOCAL_MAVEN_VERSION' in plugin.yml"
+        # Check if the version of Maven and the version of the tag equals. If not the build fails.
+        if [ $LOCAL_RELEASE_VERSION != $LOCAL_MAVEN_VERSION ]; then echo "::error::The version of the tag and the version of the pom.xml are not equal! Release is '$LOCAL_RELEASE_VERSION' and Maven is '$LOCAL_MAVEN_VERSION'."; exit 1; fi
+
+    # Set up JDK 1.8
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+        server-id: github
+        settings-path: ${{ github.workspace }}
+
+    # Build with Maven
+    - name: Build with Maven
+      run: mvn -B package --file pom.xml
+
+    # Deploy with Maven
+    - name: Publish to GitHub Packages Apache Maven
+      run: mvn deploy -s $GITHUB_WORKSPACE/settings.xml
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+
+    # Archive the artifacts from the build to the workflow run
+    - name: Archive production artifacts
+      uses: actions/upload-artifact@v1
+      with:
+        name: BetonQuest
+        path: target
+
+    # Create a GitHub release
+    - name: Create release
+      id: create_release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: ${{ github.ref }}
+        release_name: Release ${{ env.RELEASE_VERSION }}
+        draft: false
+        prerelease: false
+
+    # Upload the jar to the GitHub release
+    - name: Upload release asset
+      id: upload-release-asset
+      uses: actions/upload-release-asset@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        upload_url: ${{ steps.create_release.outputs.upload_url }} 
+        asset_path: target/BetonQuest-${{ env.RELEASE_VERSION }}.jar
+        asset_name: BetonQuest-${{ env.RELEASE_VERSION }}.jar
+        asset_content_type: application/jar
+
+    # Building the mkdocs
+    - name: Building docs
+      if: github.repository == 'BetonQuest/BetonQuest'
+      env:
+        BRANCH_NAME: echo ${{ github.ref }} | sed 's/^[^/]*\/[^/]*\///g'
+        VERSION: ${{ env.VERSION }}
+      run: |
+        bash scripts/mkdocs.sh
+
+    # Publish to discord if the branch is master && it is the BetonQuest/BetonQuest repo && github.head_ref is empty(no pull_request)
```

Comment:
I dont think its a good idea that no pull requests can exist for a release to be displayed in the discord.

Reply:
oh, this comment is out of date. there is no reference to pull requests, because release.yml is only triggered by version tags. I update this.